pilot/pkg/features/pilot.go

```diff
@@ -663,6 +663,9 @@ var (
 		"Metric scope rotation interval, set to 0 to disable the metric scope rotation").Get()
 	MetricGracefulDeletionInterval = env.Register("METRIC_GRACEFUL_DELETION_INTERVAL", 5*time.Minute,
 		"Metric expiry graceful deletion interval. No-op if METRIC_ROTATION_INTERVAL is disabled.").Get()
+
+	OptimizedConfigRebuild = env.Register("ENABLE_OPTIMIZED_CONFIG_REBUILD", false,
```

Comment:
we can default to `true`, just good to have an opt-out for possibly risky features in case they cause issues